src/app/scripts/cac/control/cac-control-directions.js

```diff
@@ -332,11 +326,18 @@ CAC.Control.Directions = (function (_, $, moment, Control, Geocoder, Routing, Ty
         typeaheadFrom.setValue(destinationText);
         typeaheadTo.setValue(originText);
 
-        // set on this object and validate
-        setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
-        setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        // have both origin and destination
+        if (directions.origin && directions.destination) {
+            // set on this object and validate
+            setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
+            setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        } else if (directions.origin) {
+            setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        } else if (directions.destination) {
+            setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
+        }
```

Comment:
I originally wrote it to only call to `planTrip` in the 'both' case, but yes, now they're independent.

Pretty sure we do want/need to `setValue`. Clearing out the `error` class for an empty field should be a special case here.